plugins/field-slider/test/field_test_helpers.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const assert = require('chai').assert;
+
+/**
+ * Assert a field's value is the same as the expected value.
+ * @param {Field} field The field.
+ * @param {*} expectedValue The expected value.
+ * @param {string=} opt_expectedText The expected text.
+ */
+function assertFieldValue(field, expectedValue, opt_expectedText) {
+  const actualValue = field.getValue();
+  const actualText = field.getText();
+  opt_expectedText = opt_expectedText || String(expectedValue);
+  assert.equal(actualValue, expectedValue, 'Value');
+  assert.equal(actualText, opt_expectedText, 'Text');
+}
+function runContructorSuiteTests(TestedField, validValueRuns, invalidValueRuns,
+    assertFieldDefault, assertField) {
+  suite('Constructor', function() {
+    test('Empty', function() {
+      const field = new TestedField();
+      assertFieldDefault(field);
+    });
+    invalidValueRuns.forEach(function(run) {
+      test(run.title, function() {
+        const field = new TestedField(...run.args);
+        assertFieldDefault(field);
+      });
+    });
+    validValueRuns.forEach(function(run) {
```

Comment:
The main difference I see between the invalid runs and the valid runs is the assertion method used, can that perhaps be passed in? 

The main difference I see between creation via constructor vs via json is the construction method. 
Can that perhaps also be included in the object definition of these tests, eg: 

What if you had something like this
```
var createWithJS = (run) => new FieldSlider(...run.args);
var createWithJson = (run) => FieldSlider.fromJson(run.json);

var assertInvalid = (field, run) => assertFieldDefault(field);
var assertValid = (field, run) => assertField(field, run.expectedValue);

var tests = {
  {title: 'Null', value: null, creation: createWithJS, assertion: assertInvalid},
  {title: 'Float', value: 1.5, expectedValue: 1.5, creation: createWithJS, assertion: assertValid},
};

runAllTests(tests);

function runAllTests(tests) {
   suite('RunAllTests', function() {
      tests.forEach(function(run) {
          test(run.title, function() {
              const field = run.creation(run);
              run.assertion(field, run);
          });
      });
   });
}

```